### PR TITLE
ref(build): Change binary name from helm to helmc

### LINKED ISSUE #453: Update build to produce and deploy `helmc` binary
Ensure generators in existing charts that recursively invoke the `helm` binary, somehow invoke `helmc` instead _without_ requiring modifications to charts.  i.e. Existing charts must not break when a user switches to `helmc`.

### PR DESCRIPTION
**Pull request by krancour**
Fixes #453 

cc @kmala

### CODE REVIEW

**Comment by mboersma** [comment]
```console
$ make native-build install
go build -o bin/helmc -ldflags "-X github.com/helm/helm-classic/cli.version=0.6.0+3b88b35" helmc.go
install -d /usr/local/bin/
install -m 755 bin/helmc /usr/local/bin/helm
```
Should that last command be renaming the binary, or should it be installed as `helmc`?

File: generator/generator.go [hunk 1]
```diff
@@ -110,6 +110,16 @@ func execute(command string, force bool) error {
 		args = args[1:]
 	}
 
+	// Templates will often include a helm:generate header such as the following:
```
Comment (kmala) [inline generator/generator.go]: do we still want the users to use helm:generate and not helmc:generate in their charts going forward?

**Comment by mboersma** [comment]
I'm having trouble because of the issue I pointed out above:
```console
$ helm generate -x manifests workflow-beta2
[ERROR] Failed to complete generation: failed to execute helm template -o
/Users/matt/.helmc/workspace/charts/workflow-beta2/manifests/deis-builder-rc.yaml -d
/Users/matt/.helmc/workspace/charts/workflow-beta2/tpl/generate_params.toml
/Users/matt/.helmc/workspace/charts/workflow-beta2/tpl/deis-builder-rc.yaml
(/Users/matt/.helmc/workspace/charts/workflow-beta2/tpl/deis-builder-rc.yaml): exec: "helmc": executable file not found in $PATH
```

**Comment by krancour** [comment]
@mboersma, nice catch.  It the install target requires an update I missed.

File: generator/generator.go (hunk 1 above)
Comment (krancour) [inline generator/generator.go]: Should still use `helm:generate`.  The "language" of the templated needs to not change at all... we want to be backwards and forwards compatible.

**Comment by krancour** [comment]
@mboersma I fixed the `install` target.

File: _scripts/ci/bintray-template.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "package": {
-    "name": "helm",
+    "name": "helmc",
     "repo": "helm-ci",
```
Comment (kmala) [inline _scripts/ci/bintray-template.json]: i had created the helmc package(https://bintray.com/deis/helm-ci/helmc/view) in the bintray....not sure if the travis automatically creates it

Comment (krancour) [inline _scripts/ci/bintray-template.json]: @kmala... I wasn't sure it was automatic either.  Thank you for doing that!

**Pull request merged**